packages/app/src/app/overmind/namespaces/editor/internalActions.ts

```diff
@@ -338,7 +322,27 @@ export const forkSandbox: AsyncAction<{
       });
     }
 
-    await actions.internal.setCurrentSandbox(forkedSandbox);
+    // When we have a server we want to set a brand new sandbox,
+    // as the preview also needs to be managed correctly
+    if (getTemplateDefinition(forkedSandbox.template).isServer) {
+      await effects.vscode.closeAllTabs();
+      actions.internal.setCurrentSandbox(forkedSandbox);
+      state.editor.modulesByPath = effects.vscode.fs.create(forkedSandbox);
```

Comment:
We can put thisline out of the if statement (happens in both)

Reply:
BElieve this is outdated?